### PR TITLE
Update GiftCardUpdate mutation

### PR DESCRIPTION
**Pull request by IKarbowiak**
Update `GiftCardUpdate` mutation.

```
type Gift  giftCardUpdate(id: ID!, input: GiftCardUpdateInput!): GiftCardUpdate

CardUpdate {
  giftCardErrors: [GiftCardError!]! @deprecated(reason: "Use errors field instead. This field will be removed in Saleor 4.0.")
  errors: [GiftCardError!]!
  giftCard: GiftCard
}

input GiftCardUpdateInput {
  tag: String
  startDate: Date
  endDate: Date
  balance: PriceInput
  expirySettings: GiftCardExpirySettingsInput
}
```

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3875 Update giftCardCreate mutation](https://app.clickup.com/t/2549495/SALEOR-3875)

File: saleor/graphql/giftcard/mutations.py [hunk 1]
```diff
@@ -87,38 +89,46 @@ class Meta:
...
 
-        cleaned_input = super().clean_input(info, instance, data)
-        cls.clean_expiry_settings(cleaned_input)
-        cls.clean_balance(cleaned_input, instance.currency)
+        cls.clean_expiry_settings(cleaned_input, instance)
+        cls.clean_balance(cleaned_input, instance)
 
-        # TODO: send an email to the customer from user_email field in SALEOR-3901
-        data.get("user_email", None)
+        cleaned_input.get("user_email", None)
```
Comment (maarcingebala) [inline saleor/graphql/giftcard/mutations.py:102]: Is this still TODO? We're not doing anything with the email.

**Review by maarcingebala — APPROVED** [review]

File: saleor/graphql/giftcard/mutations.py (hunk 1 above)
Comment (IKarbowiak) [reply saleor/graphql/giftcard/mutations.py:102]: Yes, it will be done later. It's not in scope at the moment.

Comment (maarcingebala) [reply saleor/graphql/giftcard/mutations.py:102]: 👍

**Pull request merged**